smdebug_rulesconfig/actions/actions.py

```diff
@@ -0,0 +1,177 @@
+from smdebug_rulesconfig.actions.utils import validate_string
+
+
+class Action(object):
+    def __init__(self, **action_parameters):
+        """
+        Base class for action, which is to be invoked when a rule fires.. Offers `serialize` function to convert action
+        parameters to a string dictionary. This class is not meant to be initialized directly. Accepts dictionary of
+        action parameters and drops keys whose values are `None`.
+
+        :param action_parameters: Dictionary of action parameters.
+        """
+        action_parameters["name"] = self.__class__.__name__.lower()
+        self.action_parameters = {
+            key: value for key, value in action_parameters.items() if value is not None
+        }
+
+    def serialize(self):
```

Comment:
See [this thread](https://github.com/awslabs/sagemaker-debugger-rulesconfig/pull/32#discussion_r545402799).

The following serialization is not technically valid JSON if parsed as is. It needs to be structured in a way where it will be parsed correctly (`json.loads` will eventually be called on the string within an `exec` call)